### PR TITLE
Replace reqwest with ureq (fixes #61)

### LINKED ISSUE #61: Switch to ureq http crate
As discussed in #51.

### PR DESCRIPTION
**Pull request by tuncer**

### CODE REVIEW

**Comment by ArturKovacs** [comment]
Hey @tuncer, just a reminder that this is conflicting with the main branch right now so ass soon as that's fixed, I'll review it.

**Comment by tuncer** [comment]
Done

**Comment by tuncer** [comment]
Should we embed a TODO to simplify the serde_json code once @paolobarbolini's https://github.com/algesten/ureq/pull/56 is published to crates.io? Or maybe a github ticket instead?

**Comment by ArturKovacs** [comment]
I don't think TODO should be pushed to the repository. If there are TODOs currently, that's just by mistake. So I would prefer making an issue.

**Comment by tuncer** [comment]
It's not a strict TODO like a bug but just code improvement that will be possible if and when ureq gets a release with @paolobarbolini's patch. Since it will work the same as it does now, only with one less line of code, and because we cannot predict the next ureq release, it makes no sense to block on that. That is, if you're asking.

**Comment by ArturKovacs** [comment]
I prefer issues over TODOs because it's easy to forget about TODOs. But if you would really like to add a TODO comment, then that's okay too.

>  it makes no sense to block on that

Oh don't worry we are not blocking on any issue :D I mean I'm joking of course, everything should be done with consideration. My point is that, just because there's an issue for it, in general that doesn't mean we have to wait for that issue to be comlete.

**Comment by tuncer** [comment]
Filed as #70

File: src/main.rs [hunk 1]
```diff
@@ -348,57 +346,70 @@ fn get_config_and_cache_paths() -> (PathBuf, PathBuf) {
...
-		Err(e) => {
-			println!("Could not build client for version request: {}", e);
-			return false;
+	mod errors {
+		error_chain! {
+			foreign_links {
+				Io(std::io::Error);
+				JsonError(serde_json::error::Error);
+			}
 		}
```
Comment (ArturKovacs) [inline src/main.rs:373]: You define this custom error type here, and then you use it as the return value of the `latest_release` function that you call like this:
```rust
if let Ok(info) = latest_release() {
   //...
}
```
And that's it, you don't do anything with the error. I'm not sure if this is intentional. You could print the error to the stderr or alternatively just return an `Option<ReleaseInfoJson>` from the function so you don't create a type that you end up not using.

Comment (tuncer) [reply src/main.rs:373]: Good catch, signaling of JSON errors in `check_for_updates` was lost in the process, and we're silently returning `false` right now. I will adjust `check_for_updates` accordingly.

Comment (tuncer) [reply src/main.rs:373]: Good catch, signaling of error details in `check_for_updates` was lost in the process, and we're silently returning `false` right now. I will adjust `check_for_updates` accordingly.

Comment (ArturKovacs) [reply src/main.rs:373]: I don't think that returning an error from `check_for_updates` is benefitial. We would print that error message to the stderr anyways and there's nothing else we would want to do with that error. So it's best to keep the boilerplate to a minimum and just print the error within `check_for_updates`, and return false.

Comment (tuncer) [reply src/main.rs]: Sorry for the confusion, That's not what I meant and not what I just pushed to the branch. I meant that reporting of parsing errors was lost in the migration. It's now been restored while still retaining the original signature of `check_for_updates`. The two helper functions may return errors, different types of errors, and we catch and report either error condition in `check_for_updates`.

Comment (tuncer) [reply src/main.rs]: Sorry for the confusion, that's not what I meant and not what I just pushed to the branch. I meant that reporting of parsing errors was lost in the migration, which has now been restored while still retaining the original signature of `check_for_updates`. The two helper functions may return errors, different types of errors, and we catch and report either error condition in `check_for_updates`.

File: src/main.rs [hunk 2]
```diff
@@ -381,35 +383,39 @@ mod update {
...
-						return true;
-					}
-				}
-				Err(error) => {
-					println!("Error parsing version: {}", error.to_string());
+	/// Returns true if updates are available.
+	pub fn check_for_updates() -> bool {
+		match latest_release() {
+			Ok(info) => match compare_release(&info) {
+				Ok(maybe_newer) => maybe_newer,
```
Comment (ArturKovacs) [inline src/main.rs:409]: Alright, nice! One more question: why is this called `maybe_newer`? Isn't that certain that there's a newer version available?

Comment (tuncer) [reply src/main.rs:409]: It's the boolean result of `latest > current`, which may or may not be newer (`true`). Suggestions for a better name are very welcome.

Comment (ArturKovacs) [reply src/main.rs:409]: How about `is_newer`? Because when `is_newer` is true that means that the version in `info` is newer.

Comment (tuncer) [reply src/main.rs]: Thanks, updated accordingly

**Comment by ArturKovacs** [comment]
Awesome, thanks!

**Pull request merged**